packages/workspace/src/browser/workspace-service.ts

```diff
@@ -366,11 +366,12 @@ export class WorkspaceService implements FrontendApplicationContribution {
     }
 
     /**
-     * Adds a root folder to the workspace
-     * @param uri URI of the root folder being added
+     * Adds root folder(s) to the workspace
+     * @param uris URI or URIs of the root folder(s) to add
      */
-    async addRoot(uri: URI): Promise<void> {
-        await this.spliceRoots(this._roots.length, 0, uri);
+    async addRoot(uris: URI[] | URI): Promise<void> {
+        const toAdd = Array.isArray(uris) ? uris : [uris];
+        await this.spliceRoots(this._roots.length, 0, ...toAdd);
     }
```

Comment:
I believe it can be simplified to:

```ts
async addRoot(...uris: URI[]): Promise<void> {
    await this.spliceRoots(this._roots.length, 0, ...uris);
}
```

Calls like the following are allowed:

```ts
await this.workspaceService.addRoot(...workspaceFolders);
await this.workspaceService.addRoot(uri);
```

But I won't nitpick :)